BV2-17 (Feat) : 중고거래 interface 구현

Jyuung opened the pull request:
# BV2-17 (Feat) : 중고거래 interface 구현

## 구현 목록 
- 중고거래 관련 UseCase, Command, Port 구현

Jyuung committed:
BV2-17 (Feat) : 중고거래 interface 구현

BV2-17 (Feat) : 중고거래 interface 구현

중고거래 관련 UseCase, Command, Port 구현

shinywoon commented on article/src/main/java/article/application/port/input/GetArticleUseCase.java:
```diff
@@ -0,0 +1,16 @@
+
+    ArticleSaveCommand getArticleById(Long articleId);
+
+    ArticleSaveCommand getArticleByTitle(String articleTitle);
+
+    ArticleSaveCommand getArticleByContent(String articleContent);
```
이 둘은 `like` 검색이기 때문에 Method 명에 명확히 표현해주는 것이 좋을듯 합니다.
[ex] getArticleByLikeSearch(String articleTitle);
그리고 `like` 검색 특성상 하나 이상의 Article을 반환할 확률이 높습니다.
List<AricleSaceCommand> 로 반환 받는 것이 좋을 듯 합니다.

shinywoon commented on article/src/main/java/article/application/port/input/GetArticleUseCase.java:
```diff
@@ -0,0 +1,16 @@
+
+    ArticleSaveCommand getArticleByContent(String articleContent);
+
+    List<ArticleSaveCommand> getArticleList();
```
위 두개의 기능과 중복 기능으로 판단됩니다.
List<Article> getArticleList(SearchCondition condition)

shinywoon commented on article/src/main/java/article/application/port/output/GetArticlePort.java:
```diff
@@ -0,0 +1,16 @@
+
+    Article getArticleById(String articleId);
+
+    Article getArticleByTitle(String title);
+
+    Article getArticleByContent(String content);
+
+    List<Article> getArticleList();
```
Method 통합이 가능합니다.

shinywoon reviewed (commented):
Method 명 명확성도 좋고 
코딩 실력 많이 늘었네요? ㅎㅎ
코멘트 부분 수정 후 재 PR 요청 해주세요

Jyuung committed:
BV2-17 (Refactor) OutPort 통합 및 UseCase 수정

BV2-17 (Refactor) OutPort 통합 및 UseCase 수정

1. OutPort 통합
2. 범용적으로 사용하는 ArticleSaveCommand 클래스 이름을ArticleCommand로 변경
3. ArticleSearchCommand 클래스 추가
